metadata_service/config.py

```diff
@@ -84,6 +85,10 @@ class Config:
     # format to be used in tables
     WATERMARK_DATE_FORMATS = ['%Y%m%d']
 
+    # Custom kwargs that will be passed to proxy client. Can be used to fine-tune parameters like timeout
+    # or num of retries
+    PROXY_CLIENT_KWARGS: Dict = dict()
```

Comment:
do you have example of this config? and is it possible to be leverage by other proxies?

Reply:
It definitely can be leveraged by different proxies. Two examples:

- In neo4j proxy there are settings that clearly are not set by this singleton (https://github.com/amundsen-io/amundsenmetadatalibrary/blob/master/metadata_service/proxy/neo4j_proxy.py#L54). I think that currently users don't have any way of fine tuning those. but they could be easily passed under this new configuration option.
- Some time ago we made a POC with GCP Data Catalog as Amundsen proxy and GCP client has completely different set of args in comparison to what Neo4j, Atlas or Neptune expects. We used similar solution back then.

In our case we started small with `PROXY_CLIENT_KWARGS = dict(timeout=int(os.getenv('ATLAS_REQUEST_TIMEOUT_SEC', 30)))` but there are couple of more params we could